### DISCUSSION TITLE
Welcome to check_pve Discussions!

### DISCUSSION THREAD

**Discussion by nbuchwitz** [nbuchwitz/check_pve#59] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mgualtorner** [comment]
Good morning,

I am attaching some changes that I have made to the check_pve.py application.
The changes are:

* Memory shown is equal to Memory / 100 :
** I have done this because when icinga marks a critical, the Memory is equal to the used memory plus the buffer/cache.
* For virtual machines it does not show the information of the used disk. I have modified it so that it shows the information of the used disk using the "Guest-side qemu-system agent" agent


I attach the modified function :
  def check_vm_status(self, idx: Union[str, int], **kwargs: str) -> None:
        """Check status of virtual machine by vmid or name."""
        url = self.get_url(
            "cluster/resources",
        )
        data = self.request(url, params={"type": "vm"})

        expected_state = kwargs.get("expected_state", "running")
        only_status = kwargs.get("only_status", False)

        found = False
        for vm in data:
            #bncrem mgual 20/02/2025
            mgual_vmid = vm.get("vmid", None)
            mgual_host = vm['node']
            if idx in (vm.get("name", None), vm.get("vmid", None)):
                # Check if VM (default) or LXC
                vm_type = "VM"
                if vm["type"] == "lxc":
                    vm_type = "LXC"

                if vm["status"] != expected_state:
                    self.check_message = (
                        f"{vm_type} '{vm['name']}' is {vm['status']} (expected: {expected_state})"
                    )
                    if not self.options.ignore_vm_status:
                        self.check_result = CheckState.CRITICAL
                else:
                    if self.options.node and self.options.node != vm["node"]:
                        self.check_message = (
                            f"{vm_type} '{vm['name']}' is {expected_state}, "
                            f"but located on node '{vm['node']}' instead of '{self.options.node}'"
                        )
                        self.check_result = CheckState.WARNING
                    else:
                        self.check_message = (
                            f"{vm_type} '{vm['name']}' is {expected_state} on node '{vm['node']}'"
                        )

                if vm["status"] == "running" and not only_status:
                    cpu = round(vm["cpu"] * 100, 2)
                    self.add_perfdata("cpu", cpu)

                    if self.options.values_mb:
                        memory = self.scale_value(vm["mem"])
                        # bncrem mgual maxmen Divido por 10
                        if vm_type == "VM":
                            memory= round (memory/100,2)
                        self.add_perfdata(
                            "memory",
                            memory,
                            unit=self.options.unit,
                            max=self.scale_value(vm["maxmem"]),
                        )
                        disk = self.scale_value(vm["disk"])
                        self.add_perfdata(
                            "disk",
                            disk,
                            unit=self.options.unit,
                            max=self.scale_value(vm["maxdisk"]),
                        )

                    else:
                        # bncrem mgual maxmen Divido por 10
                        memory = self.get_value(vm["mem"], vm["maxmem"])
                        if vm_type == "VM":
                            memory= round (memory/100,2)
                            mgual_url = self.get_url('nodes/'+ mgual_host  + '/qemu/' + str(mgual_vmid) + '/agent/get-fsinfo/',)
                            #mgual_data = self.request(mgual_url, params={"mountpoint": "/"})
                            mgual_data = self.request(mgual_url,)
                            for item in mgual_data['result']:
                                if item['mountpoint'] == '/':  # Verificamos si mountpoint es '/'
                                    used_bytes = item['used-bytes']
                                    total_bytes = item['total-bytes']
                            disk = self.get_value(used_bytes, total_bytes)

                        else:
                            disk = self.get_value(vm["disk"], vm["maxdisk"])
                            message=self.check_message

                        self.add_perfdata("memory", memory)
                        self.add_perfdata("disk", disk)

                    self.check_thresholds(
                            # bncrem mgual 
                        {"cpu": cpu, "memory": memory, "disk": disk}, message=self.check_message
                    )

                found = True
                break

        if not found:
            self.check_message = f"VM or LXC '{idx}' not found"
            self.check_result = CheckState.WARNING
